api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,329 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.cloud.network.vpc;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+
+import com.cloud.exception.InvalidParameterValueException;
+import com.cloud.utils.net.NetUtils;
+import com.cloud.utils.net.cidr.BadCIDRException;
+import com.cloud.utils.net.cidr.CIDR;
+
+public class OSPFZoneConfig {
+
+    private long zoneId;
+    private Protocol protocol;
+    private String ospfArea;
+    private Short helloInterval;
+    private Short deadInterval;
+    private Short retransmitInterval;
+    private Short transitDelay;
+    private Authentication authentication;
+    private String password;
+    private CIDR[] superCIDRList;
```

Comment:
Uniqness guarantee is not enough by itself, since more complex validations need to run on the CIDR list like the type and overlap , Set would not have provided a major advantage.